lib/gstreamer/va/transcoder.py

```diff
@@ -0,0 +1,80 @@
+###
+### Copyright (C) 2023 Intel Corporation
+###
+### SPDX-License-Identifier: BSD-3-Clause
+###
+
+import os
+import slash
+
+from ....lib import platform
+from ....lib.common import get_media
+from ....lib.gstreamer.transcoderbase import BaseTranscoderTest
+from ....lib.gstreamer.util import have_gst_element
+
+@slash.requires(*have_gst_element("va"))
+class TranscoderTest(BaseTranscoderTest):
+  requirements = dict(
+    decode = {
+      "avc" : dict(
+        sw = (dict(maxres = (16384, 16384)), have_gst_element("openh264dec"), "h264parse ! openh264dec ! videoconvert"),
+        hw = (platform.get_caps("decode", "avc"), have_gst_element("vah264dec"), "h264parse ! vah264dec"),
```

Comment:
This would not work because it requires a "self" instance at the time this module is loaded... that is, these "class" definitions are evaluated at module load time.  Thus, the evaluation needs to be delayed to runtime.  Currently, only string templates are supported at this level... and are later evaluated/formatted with `vars(self)` after lookup in the `get_*` base methods.

Reply:
The `f"..."` is evaluated immediately

Reply:
And unfortunately properties are not included in `vars(self)`

Reply:
Currently there is no good proposal, if there needs some time to improve, how about let this patch merged firstly, and then to give improve later? Now gst-va is mainly servicing on MTL platform, which doesn't have mulit-gpu. and we lack all transcode (avc, hevc, av1, ...) for gst-va on regular testing. @uartie @Bin-CI

Reply:
currently, there is no transcode test cases run into even single-GPU, how about let us enable this patch merge firstly, and to add more hevc, av1 transcode cases to cover single-GPU testing.
In next step, you can continue to optimize transcode on muliti-gpu.
How do you think? @uartie @Bin-CI